packages/monaco/src/browser/monaco-context-key-service.ts

```diff
@@ -58,4 +58,9 @@ export class MonacoContextKeyService extends ContextKeyService {
         return expression;
     }
 
+    parseKeys(expression: string): Set<string> {
+        const result = monaco.contextkey.ContextKeyExpr.deserialize(expression);
+        return new Set<string>(result.expr ? result.expr.map(e => e.key ? e.key : '') : []);
```

Comment:
`expr` can be recursive, so we should traverse it recursively and collect all keys

Reply:
fixed